### ISSUE TITLE
how can we calculate the test performace in S3DIS?

### ISSUE THREAD

**Issue by THHHomas** [loicland/superpoint_graph#148]
In the paper "Point Cloud Oversegmentation with Graph-Structured Deep Metric Learning", author says that The top table is for the 6-fold cross validation in S3DIS, in caption of table 1. I guess the result in table 1 is calculated like this. During train, it generate 6 weights by leave 1 out strategy, named cv1, cv2, cv3, cv4, cv5, cv6. In test, it use the 6 weights to test and get 6 accuracy, then it average the accuracy of the 6 weights. 
 But I do not know the result I test in S3DIS is not very well? Could you please tell where I am wrong? Thanks a lot.
The process I do is listed below according to S3DID.md:
1. python partition/partition.py --dataset s3dis --ROOT_PATH $S3DIS_DIR --voxel_width 0.03 --reg_strength 0.03

2. python learning/s3dis_dataset.py --S3DIS_PATH $S3DIS_DIR

3. download the weight in http://imagine.enpc.fr/~simonovm/largescale/models_s3dis.zip, and put the cv1, cv2, cv3, cv4, cv5, cv6 in results/s3dis/pretrained.
4. for FOLD in 1 2 3 4 5 6; do \
CUDA_VISIBLE_DEVICES=0 python learning/main.py --dataset s3dis --S3DIS_PATH $S3DIS_DIR --cvfold $FOLD --epochs -1 --lr_steps '[275,320]' \
--test_nth_epoch 50 --model_config 'gru_10_0,f_13' --ptn_nfeat_stn 14 --nworkers 2 --pc_attribs xyzrgbelpsvXYZ --odir "results/s3dis/pretrained/cv${FOLD}" --resume RESUME; \
done

the following is my reult:
cv6: Multisample 10: Test accuracy: 74.21582799843969, 	Test oAcc: 0.7950637256106566, 	Test avgIoU: 0.5503973431207796, 	Test mAcc: 0.6806110841096462
cv5: Multisample 10: Test accuracy: 59.55158147604431, 	Test oAcc: 0.7596150759676198, 	Test avgIoU: 0.4176217971310017, 	Test mAcc: 0.503897112117148
cv4: Multisample 10: Test accuracy: 70.8079860645853, 	Test oAcc: 0.7668289859843426, 	Test avgIoU: 0.43543434370681794, 	Test mAcc: 0.5612479585338597
cv3: Multisample 10: Test accuracy: 69.73510661683886, 	Test oAcc: 0.67198127290912, 	Test avgIoU: 0.43701311486773964, 	Test mAcc: 0.5621191724958348
cv2: Multisample 10: Test accuracy: 49.846293403215235, 	Test oAcc: 0.7151310852662777, 	Test avgIoU: 0.3254530746107008, 	Test mAcc: 0.4882944062848378
cv1: Multisample 10: Test accuracy: 64.00267244159794, 	Test oAcc: 0.7103666868767926, 	Test avgIoU: 0.45763955102741516, 	Test mAcc: 0.6088816771986811

**Comment by loicland** [comment]
Hi, sorry for the delayed response.

the procedure you used is to create the superpoint graph with handcrafted features, not the learnt ones. The results should be better though, I will run them again on my machine to check that everything works on my side.

**Comment by loicland** [comment]
Hi,

I have found an issue with pytorch 1.1. With pytorch 0.4.1 I get the wanted results but with 1.1 I getthe same poor results than you. I am investigating now.

In the meantime you can either:
- revert to 0.4.1
- retrain from scratch
- use the learned partiution.

Will let you know as soon as I find the problem.

**Comment by loicland** [comment]
I found the issue, the latest commit fixes it.

Thank you for pointing the issue out to me.

**Comment by THHHomas** [comment]
Yes, I have test the result. That's right.
